clients/azuredevopsrepo/commits_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuredevopsrepo
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+
+	"github.com/ossf/scorecard/v5/clients"
+)
+
+func Test_listStatuses(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name        string
+		ref         string
+		getRefs     fnGetRefs
+		getStatuses fnGetStatuses
+		want        []clients.Status
+		wantErr     bool
+	}{
+		{
+			name: "sha - no statuses",
+			ref:  "4b825dc642cb6eb9a060e54bf8d69288fbee4904",
+			getStatuses: func(ctx context.Context, args git.GetStatusesArgs) (*[]git.GitStatus, error) {
+				return &[]git.GitStatus{}, nil
+			},
+			want:    []clients.Status{},
+			wantErr: false,
+		},
+		{
+			name: "sha - single status",
+			ref:  "4b825dc642cb6eb9a060e54bf8d69288fbee4904",
+			getStatuses: func(ctx context.Context, args git.GetStatusesArgs) (*[]git.GitStatus, error) {
+				return &[]git.GitStatus{
+					{
+						Context: &git.GitStatusContext{
+							Name: toPtr("test"),
+						},
+						State:     toPtr(git.GitStatusStateValues.Succeeded),
+						TargetUrl: toPtr("https://example.com"),
+					},
+				}, nil
+			},
+			want: []clients.Status{
+				{
+					Context:   "test",
+					State:     "success",
+					TargetURL: "https://example.com",
+					URL:       "https://example.com",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "head - no statuses",
+			ref:  "main",
+			getRefs: func(ctx context.Context, args git.GetRefsArgs) (*git.GetRefsResponseValue, error) {
+				return &git.GetRefsResponseValue{
+					Value: []git.GitRef{
+						{
+							Statuses: &[]git.GitStatus{},
+						},
+					},
+				}, nil
+			},
+			want:    []clients.Status{},
+			wantErr: false,
+		},
+		{
+			name: "head - single status",
+			ref:  "main",
```

Comment:
technically head is "HEAD"? not sure if it matters here

Reply:
Sorry, I could've been more clear. I wasn't talking about `head` vs `HEAD`. But instead the ref is `main` instead of `HEAD`, and that gets passed into `c.listStatuses` and I thought the code does a comparison to `HEAD`, but I think it's in a different function

```go
if c.repourl.commitSHA == headCommit {
```

Reply:
Yeah, I think the solution here is actually to change the name of the test. The `head` it's referring to is `refs/heads/main`, not `HEAD`.